source/common/http/http1/codec_impl.cc

```diff
@@ -784,6 +787,9 @@ void ServerConnectionImpl::onMessageComplete() {
       Buffer::OwnedImpl buffer;
       active_request_->request_decoder_->decodeData(buffer, true);
     }
+
+    // Reset for assertion purposes.
```

Comment:
IMO reset to ensure no information from one requests persists to the next :-)
Optional, but if you're up for updating, you can do in one of your follow-ups too.

Reply:
Sure I will roll this into my next change.